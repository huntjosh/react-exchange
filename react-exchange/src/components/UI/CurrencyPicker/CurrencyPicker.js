import React, { PureComponent } from 'react';
import { Select } from 'antd';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/styles';
import ExchangeRates from '../../../api/ExchangeRates';

const { Option } = Select;

const styles = theme => ({
  currencySelect: {
    width: 100,
    margin: theme.mediumSpacing,
  },
});

class CurrencyPicker extends PureComponent {
  render() {
    const {
      defaultValue, onChange, classes, excludedCurrency,
    } = this.props;

    return (
      <Select
        defaultValue={defaultValue}
        onChange={onChange}
        className={classes.currencySelect}
      >
        {Object.values(ExchangeRates.currencies)
          .filter(currency => currency !== excludedCurrency)
          .map(currency => <Option key={currency} value={currency}>{currency}</Option>)
        }
      </Select>
    );
  }
}

CurrencyPicker.propTypes = {
  defaultValue: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  // Override Because we define this object in this file
  classes: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types,
  excludedCurrency: PropTypes.string,
};

CurrencyPicker.defaultProps = {
  defaultValue: ExchangeRates.currencies.USD,
  excludedCurrency: undefined,
};

export default withStyles(styles)(CurrencyPicker);
