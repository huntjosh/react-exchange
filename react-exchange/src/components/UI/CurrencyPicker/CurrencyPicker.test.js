import React from 'react';
import { configure, shallow } from 'enzyme/build/index';
import Adapter from 'enzyme-adapter-react-16/build/index';
import CurrencyPicker from './CurrencyPicker';

configure({ adapter: new Adapter() });
test('renders currency picker', () => {
  const component = shallow(<CurrencyPicker />);
  expect(component).toMatchSnapshot();
});