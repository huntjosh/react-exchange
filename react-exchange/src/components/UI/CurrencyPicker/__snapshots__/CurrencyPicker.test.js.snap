// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders currency picker 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Object {
          "disableGeneration": false,
          "generateClassName": [Function],
          "jss": Jss {
            "generateId": [Function],
            "id": 1,
            "options": Object {
              "Renderer": [Function],
              "createGenerateId": [Function],
              "plugins": Array [],
            },
            "plugins": PluginsRegistry {
              "plugins": Object {
                "external": Array [
                  Object {
                    "onCreateRule": [Function],
                    "onProcessStyle": [Function],
                    "onUpdate": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                    "onProcessRule": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessRule": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                ],
                "internal": Array [
                  Object {
                    "onCreateRule": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                  },
                ],
              },
              "registry": Object {
                "onChangeValue": Array [
                  [Function],
                  [Function],
                  [Function],
                ],
                "onCreateRule": Array [
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ],
                "onProcessRule": Array [
                  [Function],
                  [Function],
                ],
                "onProcessSheet": Array [],
                "onProcessStyle": Array [
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ],
                "onUpdate": Array [
                  [Function],
                ],
              },
            },
            "version": "10.0.0-alpha.5",
          },
          "sheetsCache": null,
          "sheetsManager": Map {},
          "sheetsRegistry": null,
        },
        "_currentValue2": Object {
          "disableGeneration": false,
          "generateClassName": [Function],
          "jss": Jss {
            "generateId": [Function],
            "id": 1,
            "options": Object {
              "Renderer": [Function],
              "createGenerateId": [Function],
              "plugins": Array [],
            },
            "plugins": PluginsRegistry {
              "plugins": Object {
                "external": Array [
                  Object {
                    "onCreateRule": [Function],
                    "onProcessStyle": [Function],
                    "onUpdate": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                    "onProcessRule": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessRule": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                ],
                "internal": Array [
                  Object {
                    "onCreateRule": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                  },
                  Object {
                    "onCreateRule": [Function],
                  },
                ],
              },
              "registry": Object {
                "onChangeValue": Array [
                  [Function],
                  [Function],
                  [Function],
                ],
                "onCreateRule": Array [
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ],
                "onProcessRule": Array [
                  [Function],
                  [Function],
                ],
                "onProcessSheet": Array [],
                "onProcessStyle": Array [
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ],
                "onUpdate": Array [
                  [Function],
                ],
              },
            },
            "version": "10.0.0-alpha.5",
          },
          "sheetsCache": null,
          "sheetsManager": Map {},
          "sheetsRegistry": null,
        },
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": Object {
            "disableGeneration": false,
            "generateClassName": [Function],
            "jss": Jss {
              "generateId": [Function],
              "id": 1,
              "options": Object {
                "Renderer": [Function],
                "createGenerateId": [Function],
                "plugins": Array [],
              },
              "plugins": PluginsRegistry {
                "plugins": Object {
                  "external": Array [
                    Object {
                      "onCreateRule": [Function],
                      "onProcessStyle": [Function],
                      "onUpdate": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                      "onProcessRule": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onChangeValue": [Function],
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onChangeValue": [Function],
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onChangeValue": [Function],
                      "onProcessRule": [Function],
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                  ],
                  "internal": Array [
                    Object {
                      "onCreateRule": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                    },
                  ],
                },
                "registry": Object {
                  "onChangeValue": Array [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "onCreateRule": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "onProcessRule": Array [
                    [Function],
                    [Function],
                  ],
                  "onProcessSheet": Array [],
                  "onProcessStyle": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "onUpdate": Array [
                    [Function],
                  ],
                },
              },
              "version": "10.0.0-alpha.5",
            },
            "sheetsCache": null,
            "sheetsManager": Map {},
            "sheetsRegistry": null,
          },
          "_currentValue2": Object {
            "disableGeneration": false,
            "generateClassName": [Function],
            "jss": Jss {
              "generateId": [Function],
              "id": 1,
              "options": Object {
                "Renderer": [Function],
                "createGenerateId": [Function],
                "plugins": Array [],
              },
              "plugins": PluginsRegistry {
                "plugins": Object {
                  "external": Array [
                    Object {
                      "onCreateRule": [Function],
                      "onProcessStyle": [Function],
                      "onUpdate": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                      "onProcessRule": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onChangeValue": [Function],
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onChangeValue": [Function],
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onChangeValue": [Function],
                      "onProcessRule": [Function],
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                  ],
                  "internal": Array [
                    Object {
                      "onCreateRule": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                    },
                    Object {
                      "onCreateRule": [Function],
                    },
                  ],
                },
                "registry": Object {
                  "onChangeValue": Array [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "onCreateRule": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "onProcessRule": Array [
                    [Function],
                    [Function],
                  ],
                  "onProcessSheet": Array [],
                  "onProcessStyle": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "onUpdate": Array [
                    [Function],
                  ],
                },
              },
              "version": "10.0.0-alpha.5",
            },
            "sheetsCache": null,
            "sheetsManager": Map {},
            "sheetsRegistry": null,
          },
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
